packages/os/load-tesla-kernel-modules.service.in

```diff
@@ -1,16 +1,17 @@
 [Unit]
 Description=Load additional kernel modules
 RequiresMountsFor=PREFIX/lib/modules PREFIX/src/kernels
-After=link-kernel-modules.service
-Requires=link-kernel-modules.service
+After=link-tesla-kernel-modules.service
+Requires=link-tesla-kernel-modules.service
 # Disable manual restarts to prevent loading kernel modules
 # that weren't linked by the running system
 RefuseManualStart=true
 RefuseManualStop=true
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/driverdog load-modules
+ExecCondition=/usr/bin/ghostdog match-nvidia-driver tesla
+ExecStart=/usr/bin/driverdog --modules-set nvidia-tesla load-modules
```

Comment:
Now that these units are so specific to NVIDIA, they don't really fit with the general purpose code in `os`.

I would either move them to the `kmod-*-nvidia` packages, or else turn these into unit templates and adjust the logic a bit so we can do something like:
```
[Unit]
After=link-kernel-modules@%i.service
Requires=link-kernel-modules@%i.service

...

[Service]
Type=oneshot
ExecCondition=/usr/bin/ghostdog match-driver %I
ExecStart=/usr/bin/driverdog --modules-set %I load-modules
```

Then the `kmod-*-nvidia` packages could just add symlinks to instantiate the units they supported.

Reply:
This makes sense, I'll move them into the kmod packages. I had issues getting templates to block boot so we probably will opt for explicit files here to keep things straightforward even if its a bit of copying the same things over.